src/3d/qgsvectorlayerchunkloader_p.cpp

```diff
@@ -317,8 +317,13 @@ QVector<QgsRayCastingUtils::RayHit> QgsVectorLayerChunkedEntity::rayIntersection
         QVector3D nodeIntPoint;
         int triangleIndex = -1;
 
+        // the node geometry has been translated by chunkOrigin
+        // This translation is stored in the QTransform component
+        // this needs to be taken into account to get the whole transformation
+        const QMatrix4x4 nodeTransformMatrix = node->entity()->findChild<Qt3DCore::QTransform *>()->matrix();
+        const QMatrix4x4 fullTransformMatrix = nodeTransformMatrix * transformMatrix;
```

Comment:
The `transformMatrix` is applied after `nodeTransformMatrix`, so their order should be reversed... right now it does not matter given that both are translations, but if this changes, another bug would surface if the order is not correct:
```suggestion
        const QMatrix4x4 fullTransformMatrix = transformMatrix * nodeTransformMatrix;
```

Reply:
Done.